src/System.Runtime.Extensions/src/System/IO/StreamReader.cs

```diff
@@ -39,7 +32,6 @@ public class StreamReader : TextReader
         private Decoder _decoder;
         private byte[] _byteBuffer;
         private char[] _charBuffer;
-        private byte[] _preamble;   // Encoding's preamble, which identifies this encoding.
```

Comment:
It may be possible to remove the call in `Init`, leaving a single call in `IsPreamble` (which could still be called multiple times, but I suspect multiple calls are rare). I'll play around with it tomorrow and see.